ShopAppTests/App/Account/Controllers/AccountAddressFormViewControllerSpec.swift

```diff
@@ -0,0 +1,103 @@
+//
+//  AccountAddressFormViewControllerSpec.swift
+//  ShopAppTests
+//
+//  Created by Evgeniy Antonov on 2/28/18.
+//  Copyright © 2018 RubyGarage. All rights reserved.
+//
+
+import Nimble
+import Quick
+
+import ShopApp_Gateway
+
+@testable import ShopApp
+
+class AccountAddressFormViewControllerSpec: QuickSpec {
+    override func spec() {
+        var viewController: AccountAddressFormViewController!
+        var viewModelMock: AccountAddressFormViewModelMock!
+        var viewControllerDelegateMock: AccountAddressFormControllerDelegateMock!
+        
+        beforeEach {
+            viewController = UIStoryboard(name: StoryboardNames.account, bundle: nil).instantiateViewController(withIdentifier: ControllerIdentifiers.accountAddressForm) as! AccountAddressFormViewController
+            
+            let repositoryMock = PaymentsRepositoryMock()
+            let addAddressUseCaseMock = AddAddressUseCaseMock(repository: repositoryMock)
+            let updateAddressUseCaseMock = UpdateAddressUseCaseMock(repository: repositoryMock)
+            viewModelMock = AccountAddressFormViewModelMock(addAddressUseCase: addAddressUseCaseMock, updateAddressUseCase: updateAddressUseCaseMock)
+            viewController.viewModel = viewModelMock
+            
+            viewControllerDelegateMock = AccountAddressFormControllerDelegateMock()
+            viewController.delegate = viewControllerDelegateMock
+            
+            let navigationController = NavigationController(rootViewController: UIViewController())
+            navigationController.pushViewController(viewController, animated: false)
+        }
+        
+        describe("when view loaded") {
+            beforeEach {
+                viewController.selectedAddress = Address()
+                _ = viewController.view
+            }
+            
+            it("should have a correct view model type") {
+                expect(viewController.viewModel).to(beAKindOf(AccountAddressFormViewModel.self))
+            }
+            
+            it("should have a back button") {
+                expect(viewController.navigationItem.leftBarButtonItem?.image) == #imageLiteral(resourceName: "arrow_left")
```

Comment:
I can't find where this item is set in view controller

Reply:
In BaseViewController